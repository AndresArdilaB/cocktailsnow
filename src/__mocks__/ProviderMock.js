import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { testState } from '../frontend/initialState';
import reducer from '../frontend/reducers';

const store = createStore(reducer, testState);
const history = createBrowserHistory();

const ProviderMock = ({ children }) => (
  <Provider store={store}>
    <Router history={history}>
      {children}
    </Router>
  </Provider>
);

export default ProviderMock;
