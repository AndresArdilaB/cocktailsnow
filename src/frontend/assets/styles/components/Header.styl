.header 
  align-items: center
  background: #000
  color: white
  display: flex
  height: 50px
  justify-content: center
  width: 100%


.link 
  font-size: 18px
  color: #fff

