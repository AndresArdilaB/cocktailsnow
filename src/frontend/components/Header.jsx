import React from 'react';
import { Link } from 'react-router-dom';
import '../assets/styles/components/Header.styl';

const Header = (props) => {

  return (
    <header className='header'>
      <Link to='/' className='link'>
        Home
      </Link>
    </header>
  );
};

export default Header;
