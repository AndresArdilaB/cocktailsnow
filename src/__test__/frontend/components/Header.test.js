import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../../__mocks__/ProviderMock';
import Header from '../../../frontend/components/Header';

describe('<Header />', () => {
  const header = mount(
    <ProviderMock>
      <Header />
    </ProviderMock>,
  );

  test('Header component render', () => {
    expect(header.length).toEqual(1);
  });

});

describe('Header Snapshot', () => {

  test('Testing the Header UI component', () => {
    const header = create(
      <ProviderMock>
        <Header />
      </ProviderMock>,
    );
    expect(header.toJSON()).toMatchSnapshot();
  });
});
