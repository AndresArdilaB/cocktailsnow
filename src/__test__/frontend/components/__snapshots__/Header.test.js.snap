// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot Testing the Header UI component 1`] = `
<header
  className="header"
>
  <a
    className="link"
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
</header>
`;
